pkg/ingestor/parser/cyclonedx/parser_cyclonedx.go

```diff
@@ -122,10 +129,24 @@ func (c *cyclonedxParser) addPackages(cdxBom *cdx.BOM) {
 			CPEs:     []string{comp.CPE},
 			NodeData: *assembler.NewObjectMetadata(c.doc.SourceInformation),
 		}
-		c.rootPackage.depPackages = append(c.rootPackage.depPackages, parentPackages{
+		parentPkg := parentPackages{
 			curPackage:  curPkg,
-			depPackages: []parentPackages{},
-		})
+			depPackages: []*parentPackages{},
+		}
+		c.rootPackage.depPackages = append(c.rootPackage.depPackages, &parentPkg)
+		pkgMap[comp.BOMRef] = &parentPkg
+	}
+
+	if cdxBom.Dependencies != nil {
```

Comment:
Can we also include a smaller example with dependencies that we can add to the unit test here? The above example is massive and would have tons of nodes to map out.

Reply:
done!

Reply:
Sorry for not being clear but I think need to add to `parser_cyclonedx_test.go` to use the smaller example to test out this new functionality.